codex/node.nim

```diff
@@ -514,6 +514,32 @@ proc requestStorage*(
   let purchase = await contracts.purchasing.purchase(request)
   success purchase.id
 
+proc debugDelete*(self: CodexNodeRef, cid: Cid, blockIndices: seq[int]): Future[?!int] {.async.} =
+  without manifest =? (await self.fetchManifest(cid)), err:
+    trace "Unable to fetch manifest for cid", cid, err = err.msg
+    return failure(err)
+
+  let
+    totalBlocks = manifest.blocksCount
+    treeCid = manifest.treeCid
+
+  var
+    deleted = 0
+
+  for index in blockIndices:
+    if index >= totalBlocks:
+      return failure("Index out of range. Index: " & $index & " totalBlocks: " & $totalBlocks)
```

Comment:
Hm... this leads to a strange API semantics in which if you provide an invalid block ID within a block of valid IDs, you'll delete the blocks that come before the invalid one but then get an error back. Maybe check that all of them are valid before starting to delete?

Reply:
agree

Reply:
Sure, will fix.

I do want to point out that this is designed for test scenarios. Recovery from this failure is not the plan. There should be zero tolerance for failures for the methods being used to control the scenarios. Should you get this failure, the node is in an odd state, the test has failed, and the test scenario probably needs fixing.

Reply:
Sure, I can agree with that. Still, since you've already changed it, all the better - less opportunity for behavior that could be puzzling.